operator/pkg/hook/hook.go

```diff
@@ -77,11 +81,22 @@ func (h *MutateHook) mutateHandler(cxt context.Context, req admission.Request) a
 		applicationPort = uint16(port)
 	}
 
+	labels, err := labelstool.Unmarshal(deploy.Annotations[annotationServiceLabels])
+	if err != nil {
+		h.Log.Error(err, "")
+		return errorResp(err)
+	}
+
+	aliveProbeURL := deploy.Annotations[annotationAliveProbeURLKey]
+	if aliveProbeURL == "" {
+		aliveProbeURL = defaultAliveProbeURL
+	}
+
 	service := &deploymentmodifier.MeshService{
 		Name:             deploy.Name,
```

Comment:
MeshService name isn't the name of the deployment resource, MeshService name is specified by `annotationServiceNameKey` of the annotation.

Reply:
Updated. Use name from annotation by default, otherwise use deploy.Name

Reply:
I suggest that we don't mutate deployment without configuring `annotationServiceNameKey`, because the service name is the first citizen in the EaseMesh, it should be explicitly specified by users.

Reply:
I got your idea. But it brings more complexity, the deployment will be changed if and only if it is satisfied the MutatingHookConiguration. But after filtering them twice in the annotation config, which will give a little confusing stuff.

I just explain it in another way for user experience. Do you think it is a problem?

Reply:
What's your meaning about filtering annotation config twice?

Reply:
A deployment will be injected sidecar when it satisfy:

1. MutatingHookConfiguration (namespace with label mesh.megaease.com/mesh-service)
2. Annotation has `mesh.megaease.com/service-name` (This is the second filtering😄)

Reply:
It will not confuse people. Because we must allow people to create a deployment that needn't be mutated by admission control, this is a design solely for people to control what are deployments needs to be managed by the EaseMesh.

Reply:
Okay, updated.